geoopt/optim/rwolfe.py

```diff
@@ -130,6 +138,8 @@ def phi_(self, step_size):
 
             state["new_grad"] = grad
 
+            state["der_phi"] = manifold.inner(point, -grad, state["grad_retr"])
```

Comment:
Same as comment below about reduction dimensions, you better use sum after calling norm